src/brushes/base_brush.class.js

```diff
@@ -3,122 +3,146 @@
+     * @type Number
+     * @default
+     */
+    strokeMiterLimit: 10,
+
+    /**
+     * Stroke Dash Array.
+     * @type Array
+     * @default
+     */
+    strokeDashArray: null,
+
+    /**
+     * When `true`, the free drawing is limited to the whiteboard size. Default to false.
+     * @type Boolean
+     * @default false
+     */
+
+    limitedToCanvasSize: false,
 
-    ctx.shadowColor = shadow.color;
-    ctx.shadowBlur = shadow.blur * zoom;
-    ctx.shadowOffsetX = shadow.offsetX * zoom;
-    ctx.shadowOffsetY = shadow.offsetY * zoom;
-  },
-
-  needsFullRender: function() {
-    var color = new fabric.Color(this.color);
-    return color.getAlpha() < 1 || !!this.shadow;
-  },
-
-  /**
-   * Removes brush shadow styles
-   * @private
-   */
-  _resetShadow: function() {
-    var ctx = this.canvas.contextTop;
-
-    ctx.shadowColor = '';
-    ctx.shadowBlur = ctx.shadowOffsetX = ctx.shadowOffsetY = 0;
+    /**
+     * Sets brush styles
+     * @private
+     */
+    _setBrushStyles: function () {
+      var ctx = this.canvas.contextTop;
+      ctx.strokeStyle = this.color;
+      ctx.lineWidth = this.width;
+      ctx.lineCap = this.strokeLineCap;
+      ctx.miterLimit = this.strokeMiterLimit;
+      ctx.lineJoin = this.strokeLineJoin;
+      if (fabric.StaticCanvas.supports('setLineDash')) {
+        ctx.setLineDash(this.strokeDashArray || []);
+      }
+    },
+
+    /**
+     * Sets the transformation on given context
+     * @param {RenderingContext2d} ctx context to render on
+     * @private
+     */
+    _saveAndTransform: function (ctx) {
+      var v = this.canvas.viewportTransform;
+      ctx.save();
+      ctx.transform(v[0], v[1], v[2], v[3], v[4], v[5]);
+    },
+
+    /**
+     * Sets brush shadow styles
+     * @private
+     */
+    _setShadow: function () {
+      if (!this.shadow) {
+        return;
+      }
+
+      var canvas = this.canvas,
+          shadow = this.shadow,
+          ctx = canvas.contextTop,
+          zoom = canvas.getZoom();
+      if (canvas && canvas._isRetinaScaling()) {
+        zoom *= fabric.devicePixelRatio;
+      }
+
+      ctx.shadowColor = shadow.color;
+      ctx.shadowBlur = shadow.blur * zoom;
+      ctx.shadowOffsetX = shadow.offsetX * zoom;
+      ctx.shadowOffsetY = shadow.offsetY * zoom;
+    },
+
+    needsFullRender: function () {
+      var color = new fabric.Color(this.color);
+      return color.getAlpha() < 1 || !!this.shadow;
+    },
+
+    /**
+     * Removes brush shadow styles
+     * @private
+     */
+    _resetShadow: function () {
+      var ctx = this.canvas.contextTop;
+
+      ctx.shadowColor = '';
+      ctx.shadowBlur = ctx.shadowOffsetX = ctx.shadowOffsetY = 0;
+    },
+
+    /**
+     * Removes brush shadow styles
+     * @param {Object} pointer
+     * @private
+     */
+
+    _isOutSideCanvas: function(pointer) {
+      if (pointer.x < 0 || pointer.x > this.canvas.getWidth() || pointer.y < 0 || pointer.y > this.canvas.getHeight()) {
```

Comment:
I would like to return this expression rather than having an if that return true or false.